src/auth_service/refresh_service_accounts_token/lambda_function.py

```diff
@@ -0,0 +1,169 @@
+import json
+import logging
+import os
+from typing import Dict
+
+import boto3
+from botocore.exceptions import ClientError
+
+# Initialize logger
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# Initialize AWS clients
+secrets_client = boto3.client("secretsmanager")
+lambda_client = boto3.client("lambda")
+
+# Environment variables
+LOGIN_FUNCTION_ARN = os.getenv("LOGIN_FUNCTION_ARN")
+ENVIRONMENT = os.getenv("ENVIRONMENT")
+
+# List of service accounts that need token refresh
+SERVICE_ACCOUNTS = [
+    "surveycake",
+]
+
+
+def get_secret_path(service_account: str, secret_type: str) -> str:
+    """
+    Generate secret path based on environment and service account.
+
+    Args:
+        service_account (str): Name of the service account (e.g., 'surveycake', 'slack')
+        secret_type (str): Type of secret ("password" or "access-token")
+
+    Returns:
+        str: Full secret path in Secrets Manager
+    """
+    return f"aws-educate-tpet/{ENVIRONMENT}/service-accounts/{service_account}/{secret_type}"
+
+
+def refresh_service_account_access_token(service_account: str) -> Dict:
+    """
+    Refresh access token for a specific service account.
+
+    Args:
+        service_account (str): Name of the service account to refresh token for
+
+    Returns:
+        dict: Result of the refresh operation containing:
+            - service_account: Name of the service account
+            - status: 'success' or 'failed'
+            - message: Success message or error details
+
+    Raises:
+        ClientError: If AWS service interaction fails
+        ValueError: If the login response is invalid
+    """
+    try:
+        logger.info("Starting token refresh for service account: %s", service_account)
+
+        # Retrieve service account password from Secrets Manager
+        password_response = secrets_client.get_secret_value(
+            SecretId=get_secret_path(service_account, "password")
+        )
+        password = json.loads(password_response["SecretString"])["password"]
+
+        # Prepare login payload with complete API Gateway format
+        login_payload = {
+            "headers": {"origin": "*", "Content-Type": "application/json"},
```

Comment:
Is this following PEP8?

Reply:
Hi @chungchihhan,

I've confirmed that there are no PEP8 warnings in my environment. I also followed the Claude 3.5 Sonnet to remove the unnecessary 'Origin' header, and everything looks good on my side.

![alt text](image.png)